src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/SQLiteSchema.java

```diff
@@ -0,0 +1,110 @@
+package org.openstreetmap.atlas.geography.atlas.sqlite;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Objects;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is our one-stop-shop where we set up the database schema.
+ *
+ * @author hallahan
+ */
+class SQLiteSchema
+{
+    private static final Logger logger = LoggerFactory.getLogger(SQLiteSchema.class);
+
+    // Nice to be public so you can know about what version our schema is.
+    public static final int SCHEMA_VERSION = 1;
+
+    // CACHE_SIZE is in pages, so 1,000,000 pages * 4096 bytes = 4GB
+    // Tweak this for your machine's needs.
+    public static final int CACHE_SIZE = 1_000_000;
+
+    private static final String TABLES_FILE_NAME_TEMPLATE = "org/openstreetmap/atlas/sqlite/tables_v%s.sql";
+    private static final String INDICES_FILE_NAME_TEMPLATE = "org/openstreetmap/atlas/sqlite/indices_v%s.sql";
+    private static final String USER_VERSION_TEMPLATE = "PRAGMA user_version = %s";
+    private static final String CACHE_SIZE_TEMPLATE = "PRAGMA cache_size = %s";
+
+    private final String tables;
+    private final String indices;
+
+    /**
+     * The constructor loads the schema file of the matching schema version we are using.
+     */
+    SQLiteSchema() throws IOException, NullPointerException
```

Comment:
This is pretty rare to see - throwing NPE. Thoughts on IllegalArgument instead - NPE may lead people down wrong path.